core/src/com/unciv/ui/pickerscreens/ImprovementPickerScreen.kt

```diff
@@ -72,13 +86,16 @@ class ImprovementPickerScreen(
             // canBuildImprovement() would allow e.g. great improvements thus we need to exclude them - except cancel
             if (improvement.turnsToBuild == 0 && improvement.name != Constants.cancelImprovementOrder) continue
             if (improvement.name == tileInfo.improvement) continue // also checked by canImprovementBeBuiltHere, but after more expensive tests
-            if (!tileInfo.canBuildImprovement(improvement, currentPlayerCiv)) {
-                // if there is an improvement that could remove that terrain
-                if (!tileInfoNoLast.canBuildImprovement(improvement, currentPlayerCiv)) continue
-                suggestRemoval = true
-            }
             if (!unit.canBuildImprovement(improvement)) continue
 
+            var unbuildableBecause = tileInfo.getImprovementBuildingProblems(improvement, currentPlayerCiv).toSet()
```

Comment:
improvementBuildingProblems perhaps

Reply:
Do you mean "drop `get` from the function name" or "rename the variable"? In the latter case, that sounds unnecessarily verbose for me...

Reply:
I stopped at that name too for a µs, but then liked it. There's some marginal cultural effect in which coder finds which name most intuitive. One can understand the motivation, and it's a simple one. Just my 2¢.